midend/singleArgumentSelect.cpp

```diff
@@ -2,29 +2,67 @@
 
 namespace P4 {
 
-static const IR::Expression* convertList(const IR::Expression* expression) {
+DoSingleArgumentSelect::Pair::Pair(const IR::Expression* e, const IR::Type* type) {
+    auto srcInfo = e->srcInfo;
+    if (auto m = e->to<IR::Mask>()) {
+        expr = m->left;
+        mask = m->right;
+        hasMask = true;
+    } else if (e->is<IR::DefaultExpression>()) {
+        expr = new IR::Constant(srcInfo, type, 0, 16);
+        mask = new IR::Constant(srcInfo, type, 0, 16);
+        hasMask = true;
+    } else {
+        expr = e;
+        mask = new IR::Constant(srcInfo, type, Util::maskFromSlice(type->width_bits()-1, 0), 16);
```

Comment:
What happens with zero-width bits here?

Reply:
Good point. I will add a pass to eliminate operations on zero width values.